bpf/process/types/basic.h

```diff
@@ -1506,6 +1506,12 @@ FUNC_INLINE int match_binaries(__u32 selidx, struct execve_map_value *current)
 				return 1;
 			fallthrough;
 		case op_filter_notin:
+			if (selector_options->op == op_filter_notin) {
+				if (selector_options->mbset_id != MBSET_INVALID_ID &&
+				    (current->bin.mb_bitset & (1UL << selector_options->mbset_id)))
+					return 0;
+			}
+
```

Comment:
should we add the `selector_options->mbset_id != MBSET_INVALID_ID` check the `op_filter_in` as well?

Assuming above is correct, would something like

```
		switch (selector_options->op) {
		case op_filter_in:
		case op_filter_notin:
			/* Check if we match the selector's bit in ->mb_bitset, which means that the
			 * process matches a matchBinaries section with a followChidren:true
			 * attribute either because the binary matches or because the binary of a
			 * parent matched.
			 */
			if (selector_options->mbset_id != MBSET_INVALID_ID &&
			    current->bin.mb_bitset & (1UL << selector_options->mbset_id)) {
				found_key = 0xc0ffee; /* invalid pointer, but we just check for NULL */
				break;
			}
```

work?